index.js

```diff
@@ -27,10 +27,19 @@ const getResult = async subprocess => {
 	bufferOutput(subprocess, result, 'stderr');
 
 	try {
-		const [exitCode] = await once(subprocess, 'close');
-		return {...getOutput(result), exitCode};
+		await once(subprocess, 'close');
+		return getOutput(subprocess, result);
 	} catch (error) {
-		throw Object.assign(error, getOutput(result));
+		// The `error` event on subprocess is emitted either:
+		//  - Before `spawn`, e.g. for a non-existing executable file.
+		//    Then, `subprocess.pid` is `undefined` and `close` is never emitted.
+		//  - After `spawn`, e.g. for the `signal` option.
+		//    Then, `subprocess.pid` is set and `close` is always emitted.
+		if (subprocess.pid !== undefined) {
```

Comment:
This is the relevant code: https://github.com/nodejs/node/blob/main/lib/internal/child_process.js#L396-L426